### PR TITLE
fix keypair error for debug mode and increase the wait time to avoid …

### PR DESCRIPTION
**Pull request by zhuzhih2017**
Describe the change you are making here!
1. Add more sample
2.Fix the keypair bug in debug mode
3.Increate the waiting time to avoid timeout when clean resource and binding the keypair
Please include tests. Check out these examples:

- https://github.com/hashicorp/packer/blob/master/builder/virtualbox/common/ssh_config_test.go#L19-L37
- https://github.com/hashicorp/packer/blob/master/post-processor/compress/post-processor_test.go#L153-L182

If your PR resolves any open issue(s), please indicate them like this so they will be closed when your PR is merged:

Closes #xxx
Closes #xxx

### CODE REVIEW

File: builder/alicloud/ecs/step_attach_keypair.go [hunk 1]
```diff
@@ -21,16 +22,16 @@ func (s *stepAttachKeyPar) Run(state multistep.StateBag) multistep.StepAction {
 	client := state.Get("client").(*ecs.Client)
 	config := state.Get("config").(Config)
 	instance := state.Get("instance").(*ecs.InstanceAttributesType)
-	retry_times := 3
+	start := time.Now().Add(120 * time.Second)
```
Comment (rickard-von-essen) [inline builder/alicloud/ecs/step_attach_keypair.go]: This isn't `start`. (It's start + 120 s)

**Comment by rickard-von-essen** [comment]
Maybe outside of this PR. But a nicer pattern for polling for a state is the one used in the _DigitalOcean_ builder. See (step_power_off.go L45)[https://github.com/hashicorp/packer/blob/master/builder/digitalocean/step_power_off.go#L45]

File: builder/alicloud/ecs/step_attach_keypair.go (hunk 1 above)
Comment (zhuzhih2017) [inline builder/alicloud/ecs/step_attach_keypair.go]: Thanks, I rename the variable name to timeoutPoint.

**Comment by zhuzhih2017** [comment]
@mwhooker This is a small fix for the bug in debug mode I found and add some samples, also increase the timeout value.

**Comment by mwhooker** [comment]
> Maybe outside of this PR. But a nicer pattern for polling for a state is the one used in the DigitalOcean builder. See step_power_off.go L45. Maybe that can be used for inspiration.

and I think using [Contexts](https://godoc.org/golang.org/x/net/context) would lead to a cleaner implementation as well.

for now this looks fine

**Pull request merged**